docs/docs/guides/build/components/creating-new-components/testing-your-component.md

```diff
@@ -8,92 +8,114 @@ title: Testing your component
+
+#### Loading and building definitions with `load_component_and_build_defs`
 
-`scaffold_defs_sandbox` yields an object of type `DefsPathSandbox` as a context manager. You can use the object to load the component instance and the definitions it produces.
+To test instantiation of a component, and to validate the definitions it produces, you can use the `load_component_and_build_defs` method, which loads an already-scaffolded component and builds the corresponding Definitions.
 
-For example, the following is code from our tests of our [dlt component](/guides/build/components/integrations/dlt-component-tutorial) on already-created `DefsPathSandbox`. In this case, we ensure that the definitions have loaded, and that the correct asset keys have been created:
+For example, the following is code from our tests of our [dlt component](/guides/build/components/integrations/dlt-component-tutorial). In this case, we ensure that the definitions have loaded, and that the correct asset keys have been created:
 
 ```python
-with defs_sandbox.load() as (component, defs):
-    assert isinstance(component, DltLoadCollectionComponent)
-    assert len(component.loads) == 1
-    assert defs.resolve_asset_graph().get_all_asset_keys() == {
-        AssetKey(["example", "hello_world"]),
-        AssetKey(["my_source_hello_world"]),
-    }
+def test_dlt_component():
+    with create_defs_folder_sandbox() as sandbox:
+        defs_path = sandbox.scaffold_component(component_cls=DltLoadCollectionComponent)
+        with sandbox.load_component_and_build_defs(defs_path=defs_path) as (
+            component,
+            defs,
+        ):
+            assert isinstance(component, DltLoadCollectionComponent)
+            assert len(component.loads) == 1
+            assert defs.resolve_asset_graph().get_all_asset_keys() == {
+                AssetKey(["example", "hello_world"]),
+                AssetKey(["my_source_hello_world"]),
+            }
 ```
 
-However, that is just using the default `defs.yaml` file. Usually, you will want to customize the body of `defs.yaml`. For that, there is the `component_body` argument to `load`, demonstrated in the code that tests our `PythonScriptComponent`:
+#### Testing multiple components
 
-```python
+These utilities are also useful for testing multiple components in a single test. For example, testing the `TemplatedSqlComponent` with a Snowflake connection:
 
-def test_pipes_subprocess_script_hello_world() -> None:
-    with scaffold_defs_sandbox(component_cls=PythonScriptComponent) as sandbox:
-        # Create the script we will execute
-        execute_path = sandbox.defs_folder_path / "script.py"
-        execute_path.write_text("print('hello world')")
+```python
+def test_snowflake_component():
+    with create_defs_folder_sandbox() as sandbox:
+        sandbox.scaffold_component(
+            component_cls=SqlComponent,
+            defs_path="sql_execution_component",
+            component_body={
```

Comment:
defs_body? something that indicates that this is the equivalent of `defs.yaml`

Reply:
Updated to `defs_yaml_contents`